doc: change mailing list references to github (#3082)
UIUC-PPL/charm: doc/faq/manual.rst

.. contents::
   :depth: 3

> For answers to questions not on this list, please contact us
> on our `GitHub <https://github.com/UIUC-PPL/charm>`_.

Author: ```suggestion
For answers to questions not on this list, please create an issue or discussion
on our `GitHub <https://github.com/UIUC-PPL/charm>`_.
```
